hoomd/md/BondTablePotential.h

```diff
@@ -68,6 +68,18 @@ class PYBIND11_EXPORT BondTablePotential : public ForceCompute
                           Scalar rmin,
                           Scalar rmax);
 
+    /// Set parameters from Python.
+    void setParamsPython(std::string type, pybind11::dict params);
+
+    /// Get the parameters for a particular type.
+    pybind11::dict getParams(std::string type);
+
+    /// Get the width
+    unsigned int getWidth()
+        {
+        return m_table_width;
+        }
```

Comment:
I was under the impression that it is bad form to include implementation in the header in order to not accidentally violate the "one definition rule". (I am new to C++, so this could be incorrect)

Reply:
C++ developers commonly include the implementation for short methods in the header file. This allows the compiler to inline those functions when they are called for performance. When the implementation is in the .cc file, there is no opportunity for inlining and the method must be called with all the overhead of a function call.

We add preprocessor macros near the top of the file to create an inclusion guard. This ensures that the method is only defined once.